packages/tupaia-web/src/pages/Register/Register.tsx

```diff
@@ -0,0 +1,174 @@
+                required: 'Required',
+              })}
+            />
+            <TextField
+              name="lastName"
+              label="Last name *"
+              error={!!errors?.lastName}
+              helperText={errors?.firstNalastNameme && errors?.lastName.message}
+              inputRef={register({
+                required: 'Required',
+              })}
+            />
+            <TextField
+              name="emailAddress"
+              type="email"
+              label="Email *"
+              error={!!errors?.emailAddress}
+              helperText={errors?.emailAddress && errors?.emailAddress.message}
+              inputRef={register({
+                required: 'Required',
+                ...FORM_FIELD_VALIDATION.EMAIL,
+              })}
+            />
+            <TextField
+              name="contactNumber"
+              label="Contact number (optional)"
+              error={!!errors?.contactNumber}
+              helperText={errors?.contactNumber && errors?.contactNumber.message}
+              inputRef={register()}
+            />
+            <TextField
+              name="password"
+              label="Password *"
+              type="password"
+              error={!!errors?.password}
+              helperText={errors?.password && errors?.password.message}
+              inputRef={register({
+                required: 'Required',
+                ...FORM_FIELD_VALIDATION.PASSWORD,
+              })}
+            />
+            <TextField
+              name="passwordConfirm"
+              label="Confirm password *"
+              type="password"
+              error={!!errors?.passwordConfirm}
+              helperText={errors?.passwordConfirm && errors?.passwordConfirm.message}
+              inputRef={register({
+                required: 'Required',
+                validate: value => value === getValues('password') || 'Passwords do not match.',
+                ...FORM_FIELD_VALIDATION.PASSWORD,
+              })}
+            />
+            <TextField
+              name="employer"
+              label="Employer *"
+              error={!!errors?.employer}
+              helperText={errors?.employer && errors?.employer.message}
+              inputRef={register({
+                required: 'Required',
+              })}
+            />
+            <TextField
+              name="position"
+              label="Position *"
+              error={!!errors?.position}
+              helperText={errors?.position && errors?.position.message}
+              inputRef={register({
+                required: 'Required',
+              })}
+            />
+            <StyledCheckbox
+              name="hasAgreed"
+              color="primary"
+              label="I agree to the terms and conditions"
+              error={!!errors.hasAgreed}
+              helperText={errors?.hasAgreed?.message}
+              inputRef={register({
+                required: 'Required',
```

Comment:
I wonder if we can DRY some of this up, since we are repeating required settings a lot in here. Not major though.